DataAccess/Repositories/SpecializationRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Core.Entities;
+using Core.Interfaces.Repositories;
+using DataAccess.Context;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccess.Repositories
+{
+    public class SpecializationRepository : ISpecializationRepository
+    {
+        ClinicContext _context;
+        public SpecializationRepository(ClinicContext context)
+        {
+            _context = context;
+        }
+        public Task<Specialization> AddSpecializationAsync(Specialization specialization)
```

Comment:
```suggestion
        public async Task<Specialization> AddSpecializationAsync(Specialization specialization)
```
The same thing for other async methods